src/main/java/ua/vedroid/dao3/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,211 @@
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Update " + car + " is failed", ex);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Delete of car with id="
+                    + id + " is failed", ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id, c.model, m.id AS m_id, m.name, m.country "
+                + "FROM cars c "
+                + "LEFT JOIN cars_drivers cd on c.id = cd.car_id "
+                + "LEFT JOIN drivers d on cd.driver_id = d.id "
+                + "LEFT JOIN manufacturers m on m.id = c.m_id "
+                + "WHERE d.id = ? AND c.is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(connection, resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Get cars by driver_id=" + driverId
+                    + " is failed", ex);
+        }
+    }
+
+    private void removeDrivers(Connection connection, Long carId) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Remove drivers with car_id=" + carId
+                    + " is failed", ex);
+        }
+    }
+
+    private void addDrivers(Connection connection, Car car) {
+        String query = "INSERT INTO cars_drivers(car_id, driver_id) VALUES (?, ?)";
+        for (Driver driver : car.getDrivers()) {
+            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                statement.setLong(1, car.getId());
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            } catch (SQLException ex) {
+                throw new DataProcessingException("Add driver=" + driver + " to car=" + car
+                        + " is failed", ex);
+            }
+        }
+    }
+
+    private Car getCarFromResultSet(Connection connection, ResultSet resultSet)
```

Comment:
is this the only thing this method does?

Reply:
Yes, but i can rename to `getCarWithManufacturerAndDriversFromResultSet()`

Reply:
@Vedroid thisMethod not only parsing resultSet, but also making additional request to fetch all drivers and also to parse them too. Let's divide logic of parsing result set for car and of fetching drivers for this car.
you will have two private methods:
getCarFromResultSet()
getDrivers()
and use them both in places where you were using getCarFromResultSet()